File: En Python - Interpretado/BuscarPrimeraVocal.py
#!/usr/bin/env python

def filtrar(i):
    if i != -1:
        return i
    return 45000

palabra = input("ingrese una palabra a evaluar: ")

palabra = palabra.lower()

a = filtrar(palabra.find("a"))
e = filtrar(palabra.find("e"))
i = filtrar(palabra.find("i"))
o = filtrar(palabra.find("o"))
u = filtrar(palabra.find("u"))

vocal = min(a, min(e, min(i, min(o, u))))

print("\nLa primera vocal es: {}".format(palabra[vocal]))
print("Su indice es: {}".format(vocal))

File: En Kotlin - JVM/BuscarPrimeraVocal.kt
package buscarprimeravocal

fun Int.filtrar(): Int = if(this != -1) this else 45000

fun main(args: Array<String>) {
    print("Ingrese una palabra a evaluar: ")
    var palabra = readLine() as String

    val a = palabra.indexOf("a", 0, true).filtrar()
    val e = palabra.indexOf("e", 0, true).filtrar()
    val i = palabra.indexOf("i", 0, true).filtrar()
    val o = palabra.indexOf("o", 0, true).filtrar()
    val u = palabra.indexOf("u", 0, true).filtrar()

    val vocal = minOf(a, minOf(e, minOf(i, minOf(o, u))))

    println("\nLa primera vocal es: ${palabra[vocal]}")
    println("Su indice es: $vocal")
}
